mkt/submit/views.py

```diff
@@ -102,39 +101,38 @@ def manifest(request):
     if (request.method == 'POST' and form.is_valid() and
             features_form_valid):
 
-        with transaction.atomic():
-            upload = form.cleaned_data['upload']
-            addon = Webapp.from_upload(upload, is_packaged=form.is_packaged())
+        upload = form.cleaned_data['upload']
+        addon = Webapp.from_upload(upload, is_packaged=form.is_packaged())
 
-            if form.is_packaged():
-                validation = json.loads(upload.validation)
-                escalate_prerelease_permissions(
-                    addon, validation, addon.latest_version)
+        if form.is_packaged():
+            validation = json.loads(upload.validation)
+            escalate_prerelease_permissions(
+                addon, validation, addon.latest_version)
 
-            # Set the device type.
-            for device in form.get_devices():
-                addon.addondevicetype_set.get_or_create(
-                    device_type=device.id)
+        # Set the device type.
+        for device in form.get_devices():
+            addon.addondevicetype_set.get_or_create(
+                device_type=device.id)
 
-            # Set the premium type, only bother if it's not free.
-            premium = form.get_paid()
-            if premium:
-                addon.update(premium_type=premium)
+        # Set the premium type, only bother if it's not free.
+        premium = form.get_paid()
+        if premium:
+            addon.update(premium_type=premium)
 
-            if addon.has_icon_in_manifest():
-                # Fetch the icon, do polling.
-                addon.update(icon_type='image/png')
-            else:
-                # In this case there is no need to do any polling.
-                addon.update(icon_type='')
+        if addon.has_icon_in_manifest():
+            # Fetch the icon, do polling.
+            addon.update(icon_type='image/png')
+        else:
+            # In this case there is no need to do any polling.
+            addon.update(icon_type='')
 
-            AddonUser(addon=addon, user=request.user).save()
-            # Checking it once. Checking it twice.
-            AppSubmissionChecklist.objects.create(addon=addon, terms=True,
-                                                  manifest=True, details=False)
+        AddonUser(addon=addon, user=request.user).save()
+        # Checking it once. Checking it twice.
+        AppSubmissionChecklist.objects.create(addon=addon, terms=True,
+                                              manifest=True, details=False)
 
-            # Create feature profile.
-            addon.latest_version.features.update(**features_form.cleaned_data)
+        # Create feature profile.
+        addon.latest_version.features.update(**features_form.cleaned_data)
```

Comment:
Nice work simplifying this. Are there any other places we need to adjust? I'm curious if there's any management commands that deal with icons or previews that need to trigger the tasks manually now?